### ISSUE TITLE
the design of the variable T

### ISSUE THREAD

**Issue by chen9300** [YYingDL/NSAL-DGAT#2]
Hello! I hope this message finds you well. I’m currently studying your excellent code and have a question about the design of the variable T in the training loop:

T = len(data_loader_dict["target_loader"].dataset) // args.batch_size
for i in range(T):
    _, src_examples = next(src_examples_iter)
    _, tar_examples = next(tar_examples_iter)
    ...
I noticed that in this setup, each epoch only uses a portion of the source domain data (e.g., if the target domain has 3433 samples, only 3433 source samples are selected per epoch), while the remaining source data (like 13×3433 samples) seems unused.
Could you kindly explain the rationale behind this design choice? I’m curious whether this is intended to align the number of source/target iterations, or if there are other considerations (e.g., domain alignment, computational efficiency) that motivate this approach.
Thank you so much for your time and sharing your work! I really appreciate your insights.

**Comment by YYingDL** [comment]
Thanks for your question. 
Due to the strategy of domain adaptation, we need to make sure the data sample of the source domain and the target domain are the same.

You can see the code of the domain adaptation:

def get_global_adversarial_result(self, f_s, f_t):
        f = self.grl(torch.cat((f_s, f_t), dim=0))
        d = self.global_classifiers(f)
        d_s, d_t = d.chunk(2, dim=0)
        d_label_s = torch.ones((f_s.size(0), 1)).to(f_s.device)
        d_label_t = torch.zeros((f_t.size(0), 1)).to(f_t.device)
        return 0.5 * (self.bce(d_s, d_label_s) + self.bce(d_t, d_label_t))
